fairseq/data/data_utils.py

```diff
@@ -156,7 +156,7 @@ def is_batch_full(num_tokens):
     for idx in indices:
         sample_lens.append(num_tokens_fn(idx))
         sample_len = max(sample_len, sample_lens[-1])
-        assert sample_len <= max_tokens, f"sentence at index {idx} exceeds max_tokens limit!"
+        assert sample_len <= max_tokens, "sentence at index {idx} exceeds max_tokens limit!"
```

Comment:
You're missing the format() call. Please change this to:
```
assert sample_len <= max_tokens, \
    "sentence at index {idx} exceeds max_tokens limit!".format(idx=idx)
```

Reply:
Done